opflexagent/utils/ep_managers/endpoint_file_manager.py

```diff
@@ -929,9 +929,35 @@ def _delete_endpoint_files(self, port_id, mac_exceptions=None):
         directory = os.path.dirname(self.epg_mapping_file)
         # Remove all existing EPs mapping for port_id
         for f in os.listdir(directory):
-            if (f.endswith('.' + FILE_EXTENSION) or f.endswith(
-                '.' + LBIFACE_FILE_EXTENSION)) and port_id in f:
-                if not any(x for x in mac_exceptions if x in f):
+            if f.endswith('.' + FILE_EXTENSION):
+                if f[:-3] in self.ext_seg_next_hop:
```

Comment:
This is nitpicky, but you could move the os.path.splitext to here, getting just the filename, and use that to check here, then invoke the split using the underscore below. I'm not going to hold this up because of that though.